### DISCUSSION TITLE
Open file with a specific connection

### DISCUSSION THREAD

**Discussion by mdf-ms2soft** [microsoft/azuredatastudio#15006] [Q&A]
I'd like to open a specific file via powershell, and specify the connection as well.  Is there a way to do this, it looks like I need to use `-c`, but its not well documented (that I can find). Right now, it looks like this: `-S=myServer -D=myDB  -E -c=???? -g 'myFile.sql'`

**Comment by Charles-Gagnon** [comment]
`azuredatastudio --help`

The flags relevant to a connection are bolded below. -c is for executing a command (like from the command palette), that wouldn't normally be included if you just want to open a file and connect to a server. 

Options
  -d --diff <file> <file>           Compare two files with each other.
  -a --add <folder>                 Add folder(s) to the last active window.
  -g --goto <file:line[:character]> Open a file at the path on the specified
                                    line and character position.
  -n --new-window                   Force to open a new window.
  -r --reuse-window                 Force to open a file or folder in an
                                    already opened window.
  -w --wait                         Wait for the files to be closed before
                                    returning.
  --locale <locale>                 The locale to use (e.g. en-US or zh-TW).
  --user-data-dir <dir>             Specifies the directory that user data is
                                    kept in. Can be used to open multiple
                                    distinct instances of Code.
  -h --help                         Print usage.
  -c --command <command-name>       Name of command to run
  **-D --database <database>          Database name
  -S --server <server>              Server name
  -U --user <user-name>             User name for server connection
  --aad                             Use Azure Active Directory authentication
                                    for server connection
  -E --integrated                   Use Integrated authentication for server
                                    connection**

**Comment by MichaelAllenMiller** [comment]
What steps could I use to debug why these arguments are not working for me?

This is the command I'm running in powershell:
`azuredatastudio --server SERVER_NAME --database DATABASE_NAME -E 'PATH_TO_FILE.sql';`

It seems to connect to the server in the Connections list, but the connection is not set for the query/file/window.
![image](https://user-images.githubusercontent.com/1903214/150402184-bb64b3ae-de49-4471-99dc-a7988a93497d.png)

**Reply by Charles-Gagnon** [reply]
Unfortunately, this just looks like a bug, I was able to repro it and looking at the code it does try to connect the editor, so something isn't right there. I filed an issue for us to track it: https://github.com/microsoft/azuredatastudio/issues/18132

**Reply by MichaelAllenMiller** [reply]
Thank you!
